frontend/src/model/beta/releases/EditableRelease.tsx

```diff
@@ -0,0 +1,138 @@
+import { FlattenedModelImage } from 'types/interfaces'
+import { ReleaseInterface } from 'types/types'
+import { getErrorMessage } from 'utils/fetcher'
+
+type EditableReleaseProps = {
+  release: ReleaseInterface
+}
+
+export default function EditableRelease({ release }: EditableReleaseProps) {
+  const [isEdit, setIsEdit] = useState(false)
+  const [semver, setSemver] = useState(release.semver)
+  const [releaseNotes, setReleaseNotes] = useState(release.notes)
+  const [isMinorRelease, setIsMinorRelease] = useState(!!release.minor)
+  const [artefacts, setArtefacts] = useState<File[]>([]) // TODO - Default to using the release artefact files (BAI-1026)
+  const [imageList, setImageList] = useState<FlattenedModelImage[]>(release.images)
+  const [errorMessage, setErrorMessage] = useState('')
+  const [isLoading, setIsLoading] = useState(false)
+
+  const { model, isModelLoading, isModelError } = useGetModel(release.modelId)
+  const { mutateReleases } = useGetReleasesForModelId(model?.id)
+
+  const { setUnsavedChanges } = useContext(UnsavedChangesContext)
+
+  const resetForm = useCallback(() => {
+    setSemver(release.semver)
+    setReleaseNotes(release.notes)
+    setIsMinorRelease(!!release.minor)
+    setArtefacts([]) // TODO - Reset the release artefact files (BAI-1026)
+    setImageList(release.images)
+  }, [release.images, release.minor, release.notes, release.semver])
+
+  useEffect(() => {
+    resetForm()
+  }, [resetForm])
+
+  useEffect(() => {
+    setUnsavedChanges(isEdit)
+  }, [isEdit, setUnsavedChanges])
+
+  if (isModelError) {
+    return <MessageAlert message={isModelError.info.message} severity='error' />
+  }
+
+  if (!model || isModelLoading) {
+    return <Loading />
+  }
+
+  const handleEdit = () => {
+    setIsEdit(true)
+  }
+
+  const handleCancel = () => {
+    setIsEdit(false)
+  }
+
+  const handleSubmit = async () => {
+    setIsLoading(true)
+
+    const fileIds: string[] = []
+    for (const artefact of artefacts) {
+      const postArtefactResponse = await postFile(artefact, model.id, artefact.name, artefact.type)
+      if (postArtefactResponse.ok) {
+        const res = await postArtefactResponse.json()
+        fileIds.push(res.file._id)
+      } else {
+        return setErrorMessage(await getErrorMessage(postArtefactResponse))
+      }
+    }
+
+    const updatedRelease: UpdateReleaseParams = {
+      modelId: model.id,
+      semver,
+      modelCardVersion: model.card.version,
+      notes: releaseNotes,
+      minor: isMinorRelease,
+      fileIds: fileIds,
+      images: imageList,
+    }
+
+    const response = await putRelease(updatedRelease)
+
+    if (!response.ok) {
+      const error = await getErrorMessage(response)
+      setIsLoading(false)
+      return setErrorMessage(error)
+    }
+
+    setIsLoading(false)
+    setIsEdit(false)
+    mutateReleases()
+  }
+
+  return (
+    <Box py={1}>
+      <EditableFormHeading
+        heading={
+          <div>
```

Comment:
Does this need to be a div or can we use a fragment?

Reply:
Internally the `EditableFormHeading` puts the `heading` inside of a `Stack direction='row'` so the `div` is needed so that these are rendered on top of each other instead of side-by-side. I've had to do the same for access requests too.

I thought about managing this within `EditableFormHeading`, but I decided to let us decide how the heading will look on a case-by-case basis instead.